source/fundamentals/crud/read-operations/sort.txt

```diff
@@ -4,135 +4,385 @@ Sort Results
+
+Sorting Direction
+-----------------
+
+The direction of your sort can either be **ascending** or **descending**.
+An ascending sort orders your results from smallest to largest. A
+descending sort orders your results from largest to smallest.
 
-    // define a cursor that can iterate through the results of the query
-    MongoCursor<Document> cursor = collection.find()
-        .sort(ascending("title")).iterator();
-    // print out items in alphabetical order
-    try {
-      while (cursor.hasNext()) {
-        System.out.println(cursor.next());
-      }
-    // close the cursor
-    } finally {
-      cursor.close();
-    }
+Here are some examples of data sorted in ascending order:
 
-This sample code prints the documents in ascending order on the ``title`` field as follows:
+* Numbers: 1, 2, 3, 43, 43, 55, 120
+* Dates: 1990-03-10, 1995-01-01, 2005-10-30, 2005-12-21 
+* Words (ASCII): Banana, Dill, carrot, cucumber, hummus
+
+Here are some examples of data sorted in descending order:
+
+* Numbers: 100, 30, 12, 12, 9, 3, 1
+* Dates: 2020-01-01, 1998-12-11, 1998-12-10, 1975-07-22 
+* Words (reverse ASCII): pear, grapes, apple, Cheese
+
+The following subsections show how to specify these sort criteria.
+
+Ascending
+~~~~~~~~~
+
+To specify an ascending sort, use the ``Sorts.ascending()`` static
+factory method. Pass the ``Sorts.ascending()`` method
+the name of the field you need to sort in ascending order.
+
+You can use pass the ``sort()`` method the output of the ``Sorts.ascending()``
+method to specify an ascending sort on a field as follows:
 
 .. code-block:: java
 
-    Document{{_id=6, title=A Dance with Dragons, length=1104, author=Martin}}
-    Document{{_id=3, title=Atlas Shrugged, length=1088, author=Rand}}
-    Document{{_id=5, title=Cryptonomicon, length=918, author=Stephenson}}
-    Document{{_id=4, title=Infinite Jest, length=1104, author=Wallace}}
-    Document{{_id=2, title=Les Misérables, length=1462, author=Hugo}}
-    Document{{_id=1, title=The Brothers Karamazov, length=824, author=Dostoyevsky}}
-
-Sometimes, the order of two or more documents is ambiguous using a
-specified sort. Consider the case of sorting by length: in the above case, 
-both "A Dance with Dragons" and "Infinite Jest" have ``1104`` pages, so 
-the order in which ``sort()`` returns them is not guaranteed. To resolve ties in 
-your sorted results in a repeatable way, add additional fields to the 
-``Sorts.ascending()`` function call. 
-Call the ``sort()`` method, specifying ``length`` as 
-the first field by which to sort and the ``title`` field as the second field. In the event 
-that the ``length`` field comparison results in a tie, the operation compares the tied results 
-using the ``title`` field:
+   import static com.mongodb.client.model.Sorts.ascending;
+
+   // <MongoCollection setup code here>
+
+   collection.find().sort(ascending("<field name>"));
 
+The above ``sort()`` method returns a ``FindIterable`` object containing the
```

Comment:
This could be misleading to a Java developer.  A Java developer reading that an `Iterable` contains a bunch of objects might be led to think that the objects are all in memory, when in fact the FindIterable is just a logical collection (that doesn't actually contain any documents at all in memory)  that should be streamed via a cursor.  I'm not sure how to rephrase it but maybe you can think of an alternative?  Or we can discuss on Slack.

Reply:
Thanks for the note Jeff. I agree that "containing" is misleading here. Thanks for catching this and helping me better understand cursors in MongoDB. Do you think this edit fixes the issue (I received help [on this edit from Chris Bush in this Slack thread](https://mongodb.slack.com/archives/GETCHDLCB/p1617377512125500)): https://github.com/mongodb/docs-java/pull/66/commits/d9aff574d889e5104c5dc23f6f7861a425be339e

Reply:
Yep!